snapshots/lcow/lcow.go

```diff
@@ -231,6 +231,16 @@ func (s *snapshotter) Remove(ctx context.Context, key string) error {
 	path := s.getSnapshotDir(id)
 	renamed := s.getSnapshotDir("rm-" + id)
 	if err := os.Rename(path, renamed); err != nil && !os.IsNotExist(err) {
+		// Sometimes if there are some open handles to the files (especially VHD)
+		// inside the snapshot directory the rename call will return "access
+		// denied" or "file is being used by another process" errors.  Just
+		// returning that error causes the entire snapshot garbage collection
+		// operation to fail. To avoid that we return failed pre-condition error
+		// here so that snapshot garbage collection can continue and can cleanup
+		// other snapshots.
+		if strings.Contains(err.Error(), "being used by another process") || os.IsPermission(err) {
```

Comment:
We probably can't anticipate all the errors so should we wrap all rename failures?

Reply:
Hmm I was thinking that we only want to continue the cleanup when one of these two things occur (because we hope that these errors might get resolved some time later) but we still want to fail the snapshot garbage collection if something else causes the failure. But after thinking more about this I feel that either way it is better to continue with the cleanup since that way we can at least avoid filling up the disk.